src/main/java/org/folio/rest/impl/PatronServicesResourceImpl.java

```diff
@@ -1167,4 +1177,42 @@ private String buildQueryWithRequesterId(String requesterId, String sortBy) {
     return format("(requesterId==%s and status==Open*)", requesterId);
   }
 
+  private CompletableFuture<Boolean> isEcsTlrFeatureEnabled(VertxOkapiHttpClient httpClient,
+    Map<String, String> okapiHeaders) {
+
+    return httpClient.get(ECS_TLR_SETTINGS_URL_PATH.getValue(), Map.of(), okapiHeaders)
+      .thenApply(ResponseInterpreter::extractResponseBody)
+      .thenCompose(body -> getEcsTlrFeatureValue(body, httpClient, okapiHeaders));
+  }
+
+  private CompletableFuture<Boolean> getEcsTlrFeatureValue(JsonObject body,
+    VertxOkapiHttpClient client, Map<String, String> okapiHeaders) {
+
+    return Objects.nonNull(body)
+      ? completedFuture(body.getBoolean(ECS_TLR_FEATURE_KEY))
+      : getCirculationStorageEcsTlrFeatureValue(client, okapiHeaders);
+  }
+
+  private CompletableFuture<Boolean> getCirculationStorageEcsTlrFeatureValue(
+    VertxOkapiHttpClient client, Map<String, String> okapiHeaders) {
+
+    return client.get(CIRCULATION_SETTINGS_STORAGE_URL_PATH.getValue(), Map.of(), okapiHeaders)
+      .thenApply(ResponseInterpreter::extractResponseBody)
+      .thenCompose(this::getCirculationStorageEcsTlrFeatureValue);
+  }
+
+  private CompletableFuture<Boolean> getCirculationStorageEcsTlrFeatureValue(JsonObject body) {
```

Comment:
this is a synchronous operation, no need to wrap the result into a future

Reply:
done